### PR TITLE
fix(messages): forward to inbox after deleting a message

### LINKED ISSUE #7004: Message delete forwards to 404 (referrer)
Maybe check referer value for the GUID we're deleting and instead forward to inbox.

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #7004

### CODE REVIEW

File: mod/messages/views/default/object/messages.php [hunk 1]
```diff
@@ -62,7 +62,7 @@
 ));
 
 $delete_link = elgg_view("output/url", array(
-						'href' => "action/messages/delete?guid=" . $message->getGUID(),
+						'href' => "action/messages/delete?guid=" . $message->getGUID() . "&full=$full",
```
Comment (mrclay) [inline mod/messages/views/default/object/messages.php]: $full is a bool here. Don't you have to cast to int to make this work?

Comment (mrclay) [inline mod/messages/views/default/object/messages.php]: Disregard. I just have a habit of casting bools. LGTM

Comment (jdalsem) [inline mod/messages/views/default/object/messages.php]: this is just a lazy solution :)

File: mod/messages/actions/messages/delete.php [hunk 2]
```diff
@@ -4,17 +4,23 @@
  */
 
 $guid = (int) get_input('guid');
+$full = (int) get_input('full', 0);
```
Comment (mrclay) [inline mod/messages/actions/messages/delete.php]: If this cast were (bool) it might be less confusing.

**Comment by jdalsem** [comment]
now casting to a boolean in the action

**Comment by mrclay** [comment]
LGTM

**Comment by juho-jaakkola** [comment]
LGTM

**Pull request merged**